Add Custom Line List Import API Docs (#996)
spacetelescope/jdaviz: docs/notebook/import_data.rst

    imviz = Imviz()
    imviz.app
    imviz.load_data("/path/to/data/image.fits")

Importing Custom Line Lists

> ---------------------------
> Jdaviz comes with curated line lists built from the scientific community. If you can't find the lines you need, you can add your own!::

Reviewer: ```suggestion
===========================

Jdaviz comes with curated line lists built by the scientific community. If you cannot find the lines you need, you can add your own, as follows::
```

    from astropy.table import QTable

>     import astropy.units as u

Reviewer: Nitpick.
```suggestion
    from astropy import units as u
```

>     lt = QTable()

Reviewer: `lt` is too cryptic. I propose a more legible variable name.
```suggestion
    my_line_list = QTable()
```

Author: Good point! I did make it clearer in https://github.com/spacetelescope/aas239-jwebbinar/pull/12, so probably a good idea here too

>     lt['linename'] = ['Hbeta','Halpha']

Reviewer: This is not the only way to construct a table. Maybe link to https://docs.astropy.org/en/latest/table/construct_table.html using intersphinx?

If you foresee people building a huge line list table, also link to https://docs.astropy.org/en/latest/table/index.html#performance-tips (if not, you don't have to). I'm talking about like maybe several thousand rows.

Author: Always a fan of additional documentation! I'll add those links, thanks!

Author: @pllim I never used intersphinx before (and it looks like there are a few places where links are still hardcoded). Any help to try to link to those pages would be much appreciated! `#newbie` :P

Author: Awesome! Much thanks! I got the Construct Table working, but I'm running into issues with the performance link. I found the `.. _astropy-table-performance:` header at the top of https://raw.githubusercontent.com/astropy/astropy/main/docs/table/performance.inc.rst, so I tried linking it using 
```
:ref:`<astropy:astropy-table-performance>`
```
but that didn't generate a link, and just generated raw text as "<astropy:astropy-table-performance>". Any ideas where I went wrong?

Reviewer 2: I think there's essentially a 0% chance anyone will be adding so many lines that they'll need the table performance link, if it's not working then just remove it IMO.

Author: Works for me :)

    lt['rest'] = [4851.3, 6563]*u.AA
    lt['redshift'] = u.Quantity(0.046) # Optional

    viz.load_line_list(lt)

>     viz.spectral_lines

Reviewer: ```suggestion
    viz.spectral_lines  # Briefly explain what this does in the GUI here?
```

    imviz = Imviz()
    imviz.app
    imviz.load_data("/path/to/data/image.fits")

Importing Custom Line Lists
===========================

> Jdaviz comes with curated line lists built by the scientific community. If you cannot find the lines you need, you can add your own by constructing an :ref:`astropy table <astropy:construct_table>`! For example::

Reviewer: Nitpick: It is easier for future review if you break this into mutiple lines. Also, I don't think the exclamation point is necessary.

```suggestion
Jdaviz comes with curated line lists built by the scientific community.
If you cannot find the lines you need, you can add your own by constructing
an :ref:`astropy table <astropy:construct_table>`; For example::
```
